libs/designer/src/lib/ui/panel/panelroot.tsx

```diff
@@ -192,6 +197,26 @@ export const PanelRoot = (props: PanelRootProps): JSX.Element => {
     dispatch(expandPanel());
   }, [dispatch]);
 
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const handleDeleteClick = () => setShowDeleteModal(true);
+  const handleDelete = () => {
+    // TODO: 12798935 Analytics (event logging)
+    const scopeNodeTypes = [
+      constants.NODE.TYPE.IF,
+      constants.NODE.TYPE.SWITCH,
+      constants.NODE.TYPE.FOREACH,
+      constants.NODE.TYPE.SCOPE,
+      constants.NODE.TYPE.UNTIL,
+    ];
+    if (scopeNodeTypes.includes(operationInfo?.type.toLowerCase())) {
+      const scopeId = selectedNode.split('-#')[0];
+      dispatch(deleteGraphNode({ graphId: scopeId ?? '', graphNode }));
+    } else {
+      dispatch(deleteOperation({ nodeId: selectedNode, isTrigger: isTriggerNode }));
+    }
```

Comment:
Please use the method isScopeNode to check for these types, we hsould not hard code it in the containers and also do not calculate scopeId like this, if that is hte graphId, then it should be graphnode.id

Reply:
ACK on isScopeNode - however for the scopeId, this is how it is calculated in ScopeCardNode, is there a reason why it should be different here? @preetriti1

Reply:
Actually there's no IsScopeNode function as far as I can tell

Reply:
You just need the nodeId, and that might be just outdated code and since the nodeIds do not contain # in their name it is just working since first segment is fetched, can you please just use selectedNode for your operation and see if something is failing (it should not but still try it out)